xain_fl/grpc/test_coordinator.py

```diff
@@ -29,38 +26,61 @@ def test_rendezvous_accept():
     assert result.response == RendezvousResponse.ACCEPT
 
 
-def test_rendezvous_later():
-    coordinator = Coordinator(required_participants=1)
+def test_rendezvous_later_fraction_1():
+    coordinator = Coordinator(
+        minimum_participants_in_round=1, fraction_of_participants=1.0
+    )
     coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
     result = coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant2")
 
     assert isinstance(result, coordinator_pb2.RendezvousReply)
     assert result.response == coordinator_pb2.RendezvousResponse.LATER
 
 
+def test_rendezvous_later_fraction_05():
+    coordinator = Coordinator(
+        minimum_participants_in_round=1, fraction_of_participants=0.5
+    )
+
+    # with 0.5 fraction it needs to accept at least two participants
+    coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
+    result = coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant2")
+
+    assert isinstance(result, coordinator_pb2.RendezvousReply)
+    assert result.response == coordinator_pb2.RendezvousResponse.ACCEPT
+
+    # the third participant must receive LATER RendezvousResponse
+    result = coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant3")
+
+    assert isinstance(result, coordinator_pb2.RendezvousReply)
+    assert result.response == coordinator_pb2.RendezvousResponse.LATER
+
+
 def test_heartbeat_reply():
     # test that the coordinator replies with the correct state and round number
-    coordinator = Coordinator()
+    coordinator = Coordinator(
+        minimum_participants_in_round=2, fraction_of_participants=1.0
+    )
     coordinator.on_message(coordinator_pb2.RendezvousRequest(), "participant1")
     result = coordinator.on_message(coordinator_pb2.HeartbeatRequest(), "participant1")
 
     assert isinstance(result, coordinator_pb2.HeartbeatReply)
     assert result.state == coordinator_pb2.State.STANDBY
     assert result.round == 0
 
-    # update the round and state of the coordinator and check again
-    coordinator.state = coordinator_pb2.State.ROUND
+    # update the round of the coordinator and check again
     coordinator.current_round = 10
```

Comment:
why is it OK to not also set and check `state`?

Reply:
because the reply to the `HeartbeatRequest` will depend on whether the participant has been selected or not, so even if the global `coordinator.state` is overridden, what is advertised to the participant can vary. 

basically i guess this test doesn't make sense anymore?

Reply:
**update**: test removed